openapi-cli/src/main/java/org/ballerinalang/generators/BallerinaClientGenerator.java

```diff
@@ -0,0 +1,1262 @@
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.COLON_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.COMMA_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.DOT_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.ELSE_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.EOF_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.EQUAL_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.FOREACH_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.FUNCTION_DEFINITION;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.FUNCTION_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.IF_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.IN_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.IS_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.OPEN_BRACE_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.OPEN_BRACKET_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.OPEN_PAREN_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.QUESTION_MARK_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.REMOTE_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.RETURNS_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.RIGHT_ARROW_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.SEMICOLON_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.STRING_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.STRING_LITERAL;
+import static org.ballerinalang.generators.GeneratorConstants.HTTP;
+import static org.ballerinalang.generators.GeneratorConstants.OPEN_PRAN;
+import static org.ballerinalang.generators.GeneratorUtils.buildUrl;
+import static org.ballerinalang.generators.GeneratorUtils.convertOpenAPITypeToBallerina;
+import static org.ballerinalang.generators.GeneratorUtils.escapeIdentifier;
+import static org.ballerinalang.generators.GeneratorUtils.extractReferenceType;
+import static org.ballerinalang.generators.GeneratorUtils.getBallerinaOpenApiType;
+
+/**
+ * This Util class use for generating ballerina client file according to given yaml file.
+ */
+public class BallerinaClientGenerator {
+    private static Server server;
+    private static Paths paths;
+    private static Filter filters;
+    private static List<ImportDeclarationNode> imports = new ArrayList<>();
+    private static boolean isQuery;
+
+    public static SyntaxTree generateSyntaxTree(Path definitionPath, Filter filter) throws IOException,
+            BallerinaOpenApiException {
+        imports.clear();
+        isQuery = false;
+        // Summaries OpenAPI details
+        OpenAPI openAPI = getBallerinaOpenApiType(definitionPath);
+        //Filter serverUrl
+        List<Server> servers = openAPI.getServers();
+        server = servers.get(0);
```

Comment:
If there are no servers listed ?

Reply:
Swagger provides '/' for when the contract missing the server section. In case the user is missing to fill the server section with the URLs, that time swagger parser will generate the error mentioning contract is not a valid contract.